.github/workflows/dockerimage.yml

```diff
@@ -31,16 +31,26 @@ jobs:
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v2
       - name: Login to GitHub Container Registry
+        if: ${{ github.event_name == 'push' }}
         uses: docker/login-action@v2
         with:
           registry: ${{ env.REGISTRY }}
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
-      - name: Build and push
+      - name: MAIN – Build and push
+        if: ${{ github.event_name == 'push' }}
         uses: docker/build-push-action@v4
         with:
           context: .
           file: Dockerfile
           platforms: linux/amd64,linux/arm64
           push: true
           tags: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:latest
+      - name: PR – build only
```

Comment:
What is the intention of this step? It runs the push action on pull request but does not push it?